pkg/server/server.go

```diff
@@ -1331,5 +1332,11 @@ func officialAddr(
 		return nil, errors.Errorf("hostname %q did not resolve to any addresses; listener address: %s", host, lnHost)
 	}
 
-	return util.NewUnresolvedAddr(lnAddr.Network(), net.JoinHostPort(host, lnPort)), nil
+	// cfgPort may need to be used if --advertise-port was set on the command line.
+	port := lnPort
+	if i, err := strconv.Atoi(cfgPort); err == nil && i > 0 {
```

Comment:
Are you sure that parsing the port as an integer is correct? I recall @bdarnell mentioning that ports can be non-numeric.

Reply:
Yeah, it's possible (but rarely done) to use strings like "http" in place of "80". We should probably allow anything but an empty string or literal "0" here. (although it doesn't matter much for this undocumented flag)